### ISSUE TITLE
Allow `$BUILDPLATFORM` in DL3029

### ISSUE THREAD

**Issue by StefanSchoof** [hadolint/hadolint#772]
- [ ] This is a bug report
- [x] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

docker is promoting to use `$BUILDPLATFORM` in a FROM statement: https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/

Using `$BUILDPLATFORM` in the FROM should not trigger [DL3029](https://github.com/hadolint/hadolint/wiki/DL3029), because the negative consequences that are mentions there are not valid when use `$BUILDPLATFORM`.

```
FROM --platform=$BUILDPLATFORM composer:2 AS builder
```

### Actual behavior

A `FROM --platform=$BUILDPLATFORM` produces a DL3029

### Steps to reproduce the behavior

Use `FROM --platform=$BUILDPLATFORM` and run hadolint.

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter UNKNOWN
```
(is 2.8.0, see #735)

**Dockerfile** (if relevant)

```Dockerfile
FROM --platform=$BUILDPLATFORM composer:2 AS builder
```

**Additional environment details (OS, stack version, etc.)**
